src/components/Table/components/TableRowWrapper/TableRowWrapper.tsx

```diff
@@ -0,0 +1,65 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React from 'react';
+import { TableRowContext } from '../../TableRowContext';
+import RenderRowOrNestedRow from '../RenderRowOrNestedRow/RenderRowOrNestedRow';
+import { Row, Selection, TableType } from '../../Table';
+
+type TableRowWrapperProps<T> = {
+  row: Row<T>;
+  isRowSelected: boolean;
+  onSelectionAdd: (selection: Selection) => void;
+  columnsHasNumberArr: boolean[];
+  columnsWithWidth: number[];
+  padded: boolean;
+  onSelectionChangeExist: boolean;
+  columnCount: number;
+  columns: string[];
+  fixedHeader: boolean;
+  type: TableType;
+  expanded: boolean;
+};
+
+const TableRowWrapper = <T extends Record<string, unknown>>(props: TableRowWrapperProps<T>) => {
+  const {
+    row,
+    isRowSelected,
+    onSelectionAdd,
+    padded,
+    columns,
+    fixedHeader,
+    type,
+    columnsHasNumberArr,
+    columnsWithWidth,
+    columnCount,
+    onSelectionChangeExist,
+    expanded,
+  } = props;
+
+  const tChange = React.useCallback(() => {
+    onSelectionAdd(row.id);
+  }, [onSelectionAdd, row.id]);
+
+  return (
+    <TableRowContext.Provider
+      value={{
+        row,
+        columnsHasNumberArr,
+        columnsWithWidth,
+        onSelectionChangeExist,
+        padded,
+        columns,
+        fixedHeader,
+        tChange,
+        type,
+        columnCount,
+        isRowSelected,
+        bordered: !expanded,
+      }}
+    >
+      <RenderRowOrNestedRow<T> row={row} />
+    </TableRowContext.Provider>
+  );
+};
+
+export default React.memo(TableRowWrapper) as typeof TableRowWrapper;
```

Comment:
Why use memo here where the component is exported and not in the definition of the component?

Reply:
its the same thing isn't it? Is like use HOC on index rather on the actual component it self. Easier to read and it easier to apply the `as typeof Component` at the end